core/structs-checker/structs-checker-core/src/lib.rs

```diff
@@ -1,12 +1,25 @@
+#![cfg_attr(enable_const_type_id, feature(const_type_id))]
+
 use std::any::TypeId;
 
+pub type TypeName = &'static str;
 pub type FieldName = &'static str;
 pub type EnumVariant = Option<&'static [(FieldName, TypeId)]>;
 
+pub type TypeInfo = (&'static str, [Option<TypeId>; 4]);
```

Comment:
You don't need to limit to 4 elements here. Const context allows you to produce `&'static [TypeId]`s easily which can be produced from array of any size.